### PR TITLE
feature: allow adding existing records in relation managers

### PR DESCRIPTION
**Pull request by ryangjchandler**
This is a work in progress still but one for @danharrin and @ryanscherler to track the progress of. 

This pull request aims to add support for attaching / adding existing records to a `BelongsToMany` relation through the use of a modal and a searchable select box (similar to `BelongsTo`).

In terms of the UI, there might be some things we can do better but for now it works quite well.

Still need to test it a little further and really nail down the API.

### CODE REVIEW

**Comment by danharrin** [comment]
Should we rename add to attach internally for clarity?

**Comment by ryangjchandler** [comment]
> Should we rename add to attach internally for clarity?

Yeah, probably makes more sense. Will rename all of the buttons, etc.

File: src/Resources/RelationManager/AttachRecord.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        $this->fillWithFormDefaults();
+    }
+
+    public function getForm()
+    {
+        $form = Form::make()
+            ->schema([
+                Select::make('related')
+                    ->label((string) Str::of($this->getRelationship())->singular()->title())
+                    ->placeholder('Start typing to search...')
```
Comment (danharrin) [inline src/Resources/RelationManager/AttachRecord.php:78]: Is there an existing translation string for this?

Comment (ryangjchandler) [reply src/Resources/RelationManager/AttachRecord.php:78]: Imagine so, good spot.

**Comment by ryangjchandler** [comment]
image
image

**Review by ryanscherler — APPROVED** [review]
Looks solid to me! 👍

**Comment by ryangjchandler** [comment]
There's also now an "Attach & Attach Another" button! 
image

**Review by danharrin — APPROVED** [review]

**Comment by danharrin** [comment]
Cheers Ryan!

**Pull request merged**